test/infrastructure/docker/api/v1beta1/dockercluster_types.go

```diff
@@ -53,6 +54,17 @@ type DockerClusterSpec struct {
 type DockerLoadBalancer struct {
 	// ImageMeta allows customizing the image used for the cluster load balancer.
 	ImageMeta `json:",inline"`
+
+	// UnsafeHAProxyConfigTemplateRef allows you to replace the default HAProxy config file.
+	// This field is a reference to a config map that contains the configuration template. The key of the config map should be equal to 'value'.
+	// The content of the config map will be processed and will replace the default HAProxy config file. Please use it with caution, as there are
+	// no checks to ensure the validity of the configuration. This template will support the following variables that will be passed by the controller:
+	// $IPv6 (bool) indicates if the cluster is IPv6, $FrontendControlPlanePort (string) indicates the frontend control plane port,
+	// $BackendControlPlanePort (string) indicates the backend control plane port, $BackendServers (map[string]string) indicates the backend server
+	// where the key is the server name and the value is the address. This map is dynamic and is updated every time a new control plane
+	// node is added or removed. The template will also support the JoinHostPort function to join the host and port of the backend server.
+	// +optional
+	UnsafeHAProxyConfigTemplateRef *corev1.LocalObjectReference `json:"unsafeHAProxyConfigTemplateRef,omitempty"`
```

Comment:
Re-raising this as i think it got lost.

Not sure i like the use of `Unsafe` as that sounds scary :skull:  Could we use something like `CustomHAProxyConfigTemplateRef` or `CustomLoadBalancerConfigRef` ?

Reply:
No big opinions on my side either :-) I think the unsafe comes from:

> Let's [...] make sure to clearly document the contract about it: e.g. no validation, use at your own risk, etc.

[[0]](https://github.com/kubernetes-sigs/cluster-api/issues/7684#issuecomment-1565358197)

Reply:
There are a number of areas where there is no validation, and you can do lots of damage. For example  `PreKubeadmCommands`.  

If we have a warning in the documentation of the field that states "you better know what you are doing" then that's probably enough.

Reply:
I agree with Richard. I think we don't need Unsafe in the field name

There are so many ways to misconfigure Cluster API easily and it's totally impossible for us to validate. We can't prefix everything with Unsafe